Prevent user from declaring war if it would break alliances

blabber opened the pull request:
Part of #2177.

lmoureaux reviewed (changes requested):
From reading the rest of the function, it seems that part of the code under `Check fall-out of a war declaration` is now unreachable and should be removed. This section also hints that special handling could be needed for teams.

blabber commented:
Thanks for the hint, it would indeed be sensible to not only prevent the player from breaking his own alliances, but also the alliances of their team members. I'll look I to it.

lmoureaux commented:
> Thanks for the hint, it would indeed be sensible to not only prevent the player from breaking his own alliances, but also the alliances of their team members. I'll look into it.

Are we going towards diplomacy acting at the team level instead of the nation level? This sounds like a real possibility...

blabber commented:
> Are we going towards diplomacy acting at the team level instead of the nation level? This sounds like a real possibility...

I wouldn't go that far, and I'm not sure it would be worth the effort: most LT games seem to be teamless after all.

But where it makes sense to consider the diplomatic relations of your teammates, it certainly can't hurt.

blabber commented:
I converted this PR to draft, I need to test the change in a teams environment first.

blabber commented:
This seems to work as intended. I squashed the commits and switched the PR back to "ready for review".

![team](https://github.com/longturn/freeciv21/assets/1199257/5201e9fd-ab7a-4e9c-94aa-06ea841e7ecf)

lmoureaux commented:
Hmm, how can one cancel a pact on the first encounter?

blabber commented:
"Cancel treaty" in the nations view. I guess as the Americans were allied with our team mate we had an implicit treaty?

lmoureaux commented:
Oh it was a manual action. From the screenshot I thought the two messages were generated in sequence by the game
